yamlparse-applicative/src/YamlParse/Applicative/Pretty.hs

```diff
@@ -57,29 +73,32 @@ emptyComments = Comments []
 comment :: Text -> Comments
 comment t = Comments $ map pretty $ T.lines t
 
+
+data Color = Yellow | Grey
+
 -- | Prettyprint a 'Schema'
-schemaDoc :: Schema -> Doc ()
+schemaDoc :: Schema -> Doc Color
 schemaDoc = go emptyComments
   where
-    go :: Comments -> Schema -> Doc ()
+    go :: Comments -> Schema -> Doc Color
     go cs =
       let g = go cs
           ge = go emptyComments
-          mkComment :: Doc () -> Doc ()
-          mkComment = ("# " <>)
-          mkCommentsMDoc :: Comments -> Maybe (Doc ())
+          mkComment :: Doc Color -> Doc Color
+          mkComment = (annotate Grey) . ("# " <>)
+          mkCommentsMDoc :: Comments -> Maybe (Doc Color)
           mkCommentsMDoc = \case
             Comments [] -> Nothing
             Comments l -> Just $ align $ vsep $ map mkComment l
           addMComment :: Comments -> Maybe Text -> Comments
           addMComment c = \case
             Nothing -> c
             Just t -> c <> comment t
-          e :: Doc () -> Comments -> Doc ()
+          e :: Doc Color -> Comments -> Doc Color
           e s cs' =
             case mkCommentsMDoc cs' of
               Nothing -> s
```

Comment:
You probably want to make `s` yellow here too.

Reply:
done